File: src/main/java/aurocosh/divinefavor/common/item/talismans/spell/base/SpellOptions.java
package aurocosh.divinefavor.common.item.talismans.spell.base;

import java.util.EnumSet;

public enum SpellOptions {
    ItemUseCast,
    RightClickCast,
    OnRightCastRayTraceBlock,
    OnRightCastFindTargetEntity;

    public static final EnumSet<SpellOptions> USE_CAST = EnumSet.of(SpellOptions.ItemUseCast);
    public static final EnumSet<SpellOptions> ALL_CAST = EnumSet.of(SpellOptions.ItemUseCast, SpellOptions.RightClickCast);
    public static final EnumSet<SpellOptions> RIGHT_CAST = EnumSet.of(SpellOptions.RightClickCast);
    public static final EnumSet<SpellOptions> ENTITY_CAST = EnumSet.of(SpellOptions.RightClickCast, OnRightCastFindTargetEntity);
    public static final EnumSet<SpellOptions> ALL_CAST_TRACE = EnumSet.of(SpellOptions.ItemUseCast, SpellOptions.RightClickCast, SpellOptions.OnRightCastRayTraceBlock);
    public static final EnumSet<SpellOptions> TRACE_ONLY_CAST = EnumSet.of(SpellOptions.RightClickCast, SpellOptions.OnRightCastRayTraceBlock);
}

File: src/main/kotlin/aurocosh/divinefavor/common/item/spell_talismans/base/SpellOptions.kt
package aurocosh.divinefavor.common.item.spell_talismans.base

import java.util.*

enum class SpellOptions {
    ItemUseCast,
    RightClickCast,
    OnRightCastRayTraceBlock,
    OnRightCastFindTargetEntity;

    companion object {
        val USE_CAST: EnumSet<SpellOptions> = EnumSet.of(ItemUseCast)
        val ALL_CAST: EnumSet<SpellOptions> = EnumSet.of(ItemUseCast, RightClickCast)
        val RIGHT_CAST: EnumSet<SpellOptions> = EnumSet.of(RightClickCast)
        val ENTITY_CAST: EnumSet<SpellOptions> = EnumSet.of(RightClickCast, OnRightCastFindTargetEntity)
        val ALL_CAST_TRACE: EnumSet<SpellOptions> = EnumSet.of(ItemUseCast, RightClickCast, OnRightCastRayTraceBlock)
        val TRACE_ONLY_CAST: EnumSet<SpellOptions> = EnumSet.of(RightClickCast, OnRightCastRayTraceBlock)
    }
}
